index.d.ts

```diff
@@ -0,0 +1,280 @@
+  ignoreUrls?: Array<string | RegExp>;
+  ignoreUserAgents?: Array<string | RegExp>;
+  instrument?: boolean;
+  kubernetesNamespace?: string;
+  kubernetesNodeName?: string;
+  kubernetesPodName?: string;
+  kubernetesPodUID?: string;
+  logLevel?: LogLevel;
+  logger?: Logger;
+  metricsInterval?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+  payloadLogFile?: string;
+  secretToken?: string;
+  serverTimeout?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+  serverUrl?: string;
+  serviceName?: string;
+  serviceVersion?: string;
+  sourceLinesErrorAppFrames?: number;
+  sourceLinesErrorLibraryFrames?: number;
+  sourceLinesSpanAppFrames?: number;
+  sourceLinesSpanLibraryFrames?: number;
+  stackTraceLimit?: number;
+  transactionMaxSpans?: number;
+  transactionSampleRate?: number;
+  verifyServerCert?: boolean;
+}
+
+interface CaptureErrorOptions {
+  request?: IncomingMessage;
+  response?: ServerResponse;
+  timestamp?: number;
+  handled?: boolean;
+  user?: UserObject;
+  tags?: Tags;
+  custom?: object;
+  message?: string;
+}
+
+interface Tags {
+  [key: string]: TagValue;
+}
+
+interface UserObject {
+  id?: string | number;
+  username?: string;
+  email?: string;
+}
+
+interface ParameterizedMessageObject {
+  message: string;
+  params: Array<any>;
+}
+
+interface Logger {
+  fatal (msg: string, ...args: any[]): void;
+  fatal (obj: {}, msg?: string, ...args: any[]): void;
+  error (msg: string, ...args: any[]): void;
+  error (obj: {}, msg?: string, ...args: any[]): void;
+  warn (msg: string, ...args: any[]): void;
+  warn (obj: {}, msg?: string, ...args: any[]): void;
+  info (msg: string, ...args: any[]): void;
+  info (obj: {}, msg?: string, ...args: any[]): void;
+  debug (msg: string, ...args: any[]): void;
+  debug (obj: {}, msg?: string, ...args: any[]): void;
+  trace (msg: string, ...args: any[]): void;
+  trace (obj: {}, msg?: string, ...args: any[]): void;
+  [propName: string]: any;
+}
+
+interface TransactionOptions {
+  startTime?: number;
+  childOf?: Transaction | Span | string;
+}
+
+interface SpanOptions {
+  childOf?: Transaction | Span | string;
+}
+
+type CaptureBody = 'off' | 'errors' | 'transactions' | 'all';
+type CaptureErrorLogStackTraces = 'never' | 'messages' | 'always';
+type LogLevel = 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'fatal';
+
+type CaptureErrorCallback = (err: Error | null, id: string) => void;
+type FilterFn = (payload: Payload) => Payload | boolean | void;
+type TagValue = string | number | boolean | null | undefined;
+
+type Payload = { [propName: string]: any }
+
+interface Taggable {
+  setTag (name: string, value: TagValue): boolean;
+  addTags (tags: Tags): boolean;
+}
+
+interface StartSpanFn {
+  startSpan (name?: string, type?: string, options?: SpanOptions): Span | null;
+}
+
+// Inlined from @types/aws-lambda - start
```

Comment:
Just curious, any particular reason for inlining the external types vs not referencing them directly?

Reply:
It would add an unnecessary burden on all TypeScript users as it would force them to manually install `@types/connect` and `@types/aws-lambda`. See this conversation for details: https://github.com/elastic/apm-agent-nodejs/pull/926#discussion_r267279580

Reply:
Cool that makes sense.

Reply:
FYI, I just updated it so that the inlined styles are wrapped in a namespace